cupyx/scipy/special/convex_analysis.py

```diff
@@ -0,0 +1,129 @@
+from cupy import core
+
+
+_float_preamble = '''
+#ifndef NAN
+#define NAN __int_as_float(0x7fffffff)
+#endif
+
+#ifndef INF
+#define INF __int_as_float(0x7f800000)
+#endif
+
+
+double __device__ entr(double x) {
+    if(isnan(x)) {
+        return NAN;
+    }
+    else if(x > 0){
+        return -x * log(x);
+    } else if(x == 0){
+        return 0;
+    } else {
+        return -INF;
+    }
+}
+
+double __device__ kl_div(double x, double y) {
+    if (isnan(x) | isnan(y)) {
```

Comment:
```suggestion
    if (isnan(x) || isnan(y)) {
```
Please fix other places as well (including `&` -> `&&`)